JustSaying/AwsTools/MessageHandling/SqsNotificationListener.cs

```diff
@@ -86,39 +85,30 @@ public SqsNotificationListener WithMessageProcessingStrategy(IMessageProcessingS
             _handlerMap.Add(typeof(T), handlerFunc);
         }
 
-        public void Listen()
+        public void Listen(CancellationToken cancellationToken)
         {
             var queue = _queue.QueueName;
             var region = _queue.Region.SystemName;
             var queueInfo = $"Queue: {queue}, Region: {region}";
 
-            _cts = new CancellationTokenSource();
-            Task.Factory.StartNew(async () =>
-                {
-                    while (!_cts.IsCancellationRequested)
-                    {
-                        await ListenLoop(_cts.Token).ConfigureAwait(false);
-                    }
-                })
+            Task.Factory.StartNew(async () => { await ListenLoop(cancellationToken).ConfigureAwait(false); },
```

Comment:
This block looks like it might need some TLC with regards to the recommendations in the doc David Fowler shared as well, due to the `ContinueWith()` usage. We could do that separately though.

Reply:
Issue https://github.com/justeat/JustSaying/issues/404 created for this...hopefully you can find it :trollface:

Reply:
Is the `= default` needed? It is going to be quite unusual to use this class directly, so we don't have to concentrate on API ergonomics.

Reply:
There should be some tests proving that this code does what we expect it to do.

Reply:
Cancelling should work in our case though, because the cancellation token is passed into the `ListenLoop` method. It's just the `LogTaskEndState` method which will not work properly.